src/ts/components/subtitlesettingtoggle.ts

```diff
@@ -0,0 +1,130 @@
+import {ToggleButton, ToggleButtonConfig} from './togglebutton';
+import {Button, ButtonConfig} from './button';
+import {Container, ContainerConfig} from './container';
+import {Label, LabelConfig} from './label';
+import {SettingsPanel} from './settingspanel';
+import {SubtitleSettingsPanel} from './subtitlesettingspanel';
+import {UIInstanceManager} from '../uimanager';
+import {DOM} from '../dom';
+import {EventDispatcher, Event, NoArgs} from '../eventdispatcher';
+
+export interface SubtitleSettingsButtonConfig extends ButtonConfig {
+  subtitleSettingsPanel: SubtitleSettingsPanel;
+  settingsPanel: SettingsPanel;
+}
+
+/**
+ * A button that toggles the option menu for subtitles
+ */
+export class SubtitleSettingsOpener extends Button<ButtonConfig> {
```

Comment:
Rename to `SubtitleSettingsOpenButton` and extract to `subtitlesettingsopenbutton.ts`. To use the `SubtitleSettingsButtonConfig` for both buttons, you need to add an `abstract class SubtitleSettingsButton` inside `subtitlesettingsbutton.ts` and use that as a base class for both buttons.

General rule: Every component gets its own file (except subcomponents only used within the component where it is defined)

General rule: Components are named with their base component name as suffix, i.e. every button should be suffixed with `Button`